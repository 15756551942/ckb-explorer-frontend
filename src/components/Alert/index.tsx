import React, { useEffect } from 'react'
import { useAppState, useDispatch } from '../../contexts/providers'
import { AlertPanel } from './styled'
import i18n from '../../utils/i18n'
import { parseSimpleDateNoSecond } from '../../utils/date'
import SimpleButton from '../SimpleButton'
import { ComponentActions } from '../../contexts/actions'
import { CachedKeys } from '../../utils/const'

const Alert = () => {
  const dispatch = useDispatch()
  const {
    app: { appErrors },
    components: { maintenanceAlertVisible },
  } = useAppState()
  const [startTime, endTime] = appErrors[2].message

  const hideAlert = () => {
    sessionStorage.setItem(CachedKeys.MaintenanceAlert, 'hide')
    dispatch({
      type: ComponentActions.UpdateMaintenanceAlertVisible,
      payload: {
        maintenanceAlertVisible: false,
      },
    })
  }

  useEffect(() => {
    const hideMaintenance = sessionStorage.getItem(CachedKeys.MaintenanceAlert) === 'hide'
    if (startTime && endTime && !hideMaintenance) {
      dispatch({
        type: ComponentActions.UpdateMaintenanceAlertVisible,
        payload: {
          maintenanceAlertVisible: true,
        },
      })
    }
  }, [startTime, endTime, dispatch])

  return maintenanceAlertVisible ? (
    <AlertPanel>
      <div>
        <span>
          {i18n.t('toast.maintenance', {
            start: parseSimpleDateNoSecond(startTime, '-', false),
            end: parseSimpleDateNoSecond(endTime, '-', false),
          })}
        </span>
        <div>
          <SimpleButton className="alert__dismiss" onClick={() => hideAlert()}>
            {i18n.t('toast.dismiss')}
          </SimpleButton>
        </div>
      </div>
    </AlertPanel>
  ) : null
}

export default Alert
