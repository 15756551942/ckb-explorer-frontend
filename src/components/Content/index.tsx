import React from 'react'
import styled from 'styled-components'

const parseMargin = (width: number) => {
  if (width >= 1200) {
    return 80
  }
  return 90 / (width / 1200)
}

const ContentDiv = styled.div`
  width: 100%;
  overflow-x: hidden;
  flex: 1;
  margin-top: ${(props: { width: number }) => `${parseMargin(props.width)}px`};

  @media (max-width: 700px) {
    margin-top: 42px;
  }
`
export default ({ children, style }: { children: any; style?: any }) => {
  return (
    <ContentDiv style={style} width={window.innerWidth}>
      {children}
    </ContentDiv>
  )
}
