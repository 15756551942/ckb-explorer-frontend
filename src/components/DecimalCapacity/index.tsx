import React from 'react'
import styled from 'styled-components'
import i18n from '../../utils/i18n'

const DecimalPanel = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;

  font-family: source-code-pro, Menlo, Monaco, Consolas, Courier New, monospace;

  .decimal__zeros {
    margin-bottom: 1px;
  }

  .decimal__unit {
    margin-left: 5px;

    @media (max-width: 700px) {
      margin-bottom: 0px;
    }
  }
`

const DecimalPartPanel = styled.div`
  margin-bottom: 1px;
  font-size: ${(props: { fontSize?: string; color?: string }) => (props.fontSize ? props.fontSize : '12px')}
  color: ${(props: { fontSize?: string; color?: string }) => (props.color ? props.color : '#999999')}

  @media (max-width: 1000px) {
    font-size: ${(props: { fontSize?: string; color?: string }) => (props.fontSize ? props.fontSize : '11px')}
  }

  @media (max-width: 700px) {
    margin-bottom: 0px;
  }
`

const DecimalZerosPanel = styled.div`
  margin-bottom: 1px;
  font-size: ${(props: { fontSize?: string; color?: string }) => (props.fontSize ? props.fontSize : '12px')}
  color: ${(props: { fontSize?: string; color?: string }) => (props.color ? props.color : '#999999')}

  @media (max-width: 1000px) {
    font-size: ${(props: { fontSize?: string; color?: string }) => (props.fontSize ? props.fontSize : '11px')}
  }

  @media (max-width: 700px) {
    margin-bottom: 0px;
  }
`

export default ({
  value,
  fontSize,
  color,
  hideUnit,
}: {
  value: string
  fontSize?: string
  color?: string
  hideUnit?: boolean
}) => {
  const integer = value.split('.')[0] || '0'
  let decimal = value.split('.')[1] || ''
  let zeros = ''

  if (decimal.length < 8) {
    zeros = '0'.repeat(8 - decimal.length)
  }
  if (decimal.length === 0) {
    zeros = `.${'0'.repeat(8)}`
  } else if (decimal.length < 8) {
    zeros = '0'.repeat(8 - decimal.length)
  }
  decimal = decimal.length > 0 ? `.${decimal}` : ''

  return (
    <DecimalPanel>
      <span>{integer}</span>
      <DecimalPartPanel fontSize={fontSize} color={color}>
        {decimal}
      </DecimalPartPanel>
      <DecimalZerosPanel fontSize={fontSize} color={color}>
        {zeros}
      </DecimalZerosPanel>
      {!hideUnit && <div className="decimal__unit">{i18n.t('common.ckb_unit')}</div>}
    </DecimalPanel>
  )
}
