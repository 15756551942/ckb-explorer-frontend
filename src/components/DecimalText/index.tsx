import React from 'react'
import styled from 'styled-components'

export interface DecimalParams {
  integerFontSize?: string
  integerColor?: string
  decimalFontSize?: string
  decimalColor?: string
}

const DecimalPanel = styled.div`
  display: inner-block;
`

export default ({ value, decimalParams }: { value: string; decimalParams?: DecimalParams }) => {
  const integer = value.split('.')[0] || ''
  let decimal = value.split('.')[1] || ''
  decimal = decimal.length > 0 ? `.${decimal}` : ''
  return (
    <DecimalPanel>
      <span
        style={{
          fontSize: decimalParams ? decimalParams.integerFontSize : 'inherit',
          color: decimalParams ? decimalParams.integerColor : 'inherit',
        }}
      >
        {integer}
      </span>
      <span
        style={{
          fontSize: decimalParams ? decimalParams.decimalFontSize : 'inherit',
          color: decimalParams ? decimalParams.decimalColor : 'inherit',
        }}
      >
        {decimal}
      </span>
    </DecimalPanel>
  )
}
