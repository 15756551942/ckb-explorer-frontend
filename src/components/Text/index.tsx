import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

export const HighLightPanel = styled.div`
  color: ${props => props.theme.primary};
  font-size: 13px;
  height: 16px;

  a {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace;
    color: ${props => props.theme.primary};
  }

  a:hover {
    color: ${props => props.theme.primary};
  }
`

export const HighLightLink = ({ value, to }: { value: string; to: string }) => {
  return (
    <HighLightPanel>
      <Link to={to}>{value}</Link>
    </HighLightPanel>
  )
}
