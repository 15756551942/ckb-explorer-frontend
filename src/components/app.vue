<template>
  <div class="uk-container">
    <ul class="uk-tab uk-child-width-expand">
      <li><router-link to="/">Home</router-link></li>
      <li><router-link to="/send_transaction">Send Transaction</router-link></li>
      <li><router-link to="/blocks">Blocks</router-link></li>
      <li><router-link to="/transactions">Transactions</router-link></li>
      <li>
        <div class="uk-inline">
          <span class="uk-form-icon" uk-icon="icon: server"></span>
          <input v-model="addr" v-on:keyup.enter="connect" type="text" class="uk-input"/>
        </div>
      </li>
    </ul>

    <router-view
      :blocks="blocks"
      :transactions="transactions"
      :block="block"
      :transaction="transaction"
      :get_block="get_block"
      :get_transaction="get_transaction"
      :connect="connect"
      v-on:send_transaction="send_transaction">
    </router-view>
  </div>
</template>

<script>
import _ from 'lodash'
import JsonRPC from 'simple-jsonrpc-js'

export default {
  data: function () {
    return {
      addr: 'http://localhost:8114/',
      blocks: [],
      transactions: [],
      block: {},
      transaction: {}
    }
  },
  methods: {
    connect: function () {
      this.jrpc().call('get_tip_header', []).then((result) => {
        this.blocks = []
        this.transactions = []
        _.range(result.number, result.number - 10, -1).forEach((number, _) => {
          this.jrpc().call('get_block_hash', [number]).then((hash) => {
            this.jrpc().call('get_block', [hash]).then((block) => {
              this.blocks.push(block)
              this.transactions = this.transactions.concat(block.transactions)
            })
          })
        })
      })
    },

    get_block: function (hash) {
      this.jrpc().call('get_block', [hash]).then((block) => {
        this.block = block
        console.log(block)
      })
    },

    get_transaction: function (hash) {
      this.jrpc().call('get_transaction', [hash]).then((transaction) => {
        this.transaction = transaction
        console.log(transaction)
      })
    },

    send_transaction (tx) {
      this.jrpc().call('send_transaction', [tx]).then((result) => {
        console.log(result)
      })
    },

    jrpc: function () {
      let jrpc = new JsonRPC()
      let addr = this.addr

      jrpc.toStream = function (msg) {
        let xhr = new XMLHttpRequest()
        xhr.onreadystatechange = function () {
          if (this.readyState !== 4) return

          try {
            JSON.parse(this.responseText)
            jrpc.messageHandler(this.responseText)
          } catch (e) {
            console.error(e)
          }
        }

        xhr.open('POST', addr, true)
        xhr.setRequestHeader('Content-type', 'application/json')
        xhr.send(msg)
      }
      return jrpc
    }
  }
}
</script>
