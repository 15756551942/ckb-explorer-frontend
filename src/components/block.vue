<template>
  <table class="uk-table">
    <caption>Block #{{ block.header.number }}</caption>
    <tbody>
        <tr>
          <td>Hash</td>
          <td>#{{ block.hash }}</td>
        </tr>
        <tr>
          <td>Pre Hash</td>
          <td><router-link v-bind:to="{ name: 'blocks', params: { id: block.header.parent_hash }}">{{ block.header.parent_hash}}</router-link></td>
        </tr>
        <tr>
          <td>Timestamp</td>
          <td>{{ new Date(block.header.timestamp) | moment("YYYY-MM-DD HH:mm:ss") }}</td>
        </tr>
        <tr>
          <td>Height</td>
          <td>{{ block.header.number }}</td>
        </tr>
        <tr>
          <td>Transactions Commit</td>
          <td>{{ block.header.txs_commit }}</td>
        </tr>
        <tr>
          <td>Difficulty</td>
          <td>{{ block.header.difficulty }}</td>
        </tr>
        <tr>
          <td>Challenge</td>
          <td>{{ block.header.challenge }}</td>
        </tr>
        <tr>
          <td></td>
          <td></td>
        </tr>
    </tbody>
  </table>
</template>

<script>
export default {
  props: ['block', 'get_block'],

  created () {
    this.get_block(this.$route.params.id)
  },

  beforeRouteUpdate (to, from, next) {
    this.get_block(to.params.id)
  }
}
</script>
