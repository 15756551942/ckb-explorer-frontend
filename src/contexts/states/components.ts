export const initComponents: State.Components = {
  searchBarEditable: false,
  mobileMenuVisible: false,
  headerSearchBarVisible: false,
  filterNoResult: false,
}

export default initComponents
