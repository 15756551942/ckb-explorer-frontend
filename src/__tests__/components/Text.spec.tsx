import React, { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { HighLightLink } from '../../components/Text'
import { BrowserRouter } from 'react-router-dom'

describe('Text Component', () => {
  let component: ReactElement

  beforeAll(() => {
    component = (
      <BrowserRouter>
        <HighLightLink value="value" to="to" tooltip="tooltip" />
      </BrowserRouter>
    )
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find(HighLightLink)).toHaveLength(1)
  })
})
