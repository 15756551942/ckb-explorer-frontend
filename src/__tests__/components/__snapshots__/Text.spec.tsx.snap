// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component shallow renders 1`] = `
<div
  className="sc-AxirZ giAbtg"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    className="undefined monospace"
    href="/to"
    onClick={[Function]}
  >
    value
  </a>
</div>
`;
